reframe/core/schedulers/slurm.py

```diff
@@ -227,7 +230,22 @@ def emit_preamble(self, job):
 
     def submit(self, job):
         cmd = f'sbatch {job.script_filename}'
-        completed = _run_strict(cmd, timeout=self._submit_timeout)
+        intervals = itertools.cycle([1, 2, 3])
+        while True:
+            try:
+                completed = _run_strict(cmd, timeout=self._submit_timeout)
+                break
+            except SpawnedProcessError as e:
+                sbatch_error = 'sbatch: error: QOSMaxSubmitJobPerUserLimit'
+                if (
+                    not self._block_submission or
+                    e.exitcode != 1 or
+                    sbatch_error not in e.stderr
+                ):
+                    raise e
```

Comment:
```suggestion
                sbatch_error = 'sbatch: error: QOSMaxSubmitJobPerUserLimit'
                if (not self._block_submission or
                    sbatch_error not in e.stderr):
                    raise
```

- The `SpawnedProcessError` will be raised only if the exit code is non-zero, so this check is superfluous.
- Perhaps a regex matching instead of the `in` check would be better and less strict.